controllers/controllers.go

```diff
@@ -287,3 +296,39 @@ func markAsFreeOnSuccess(c client.Client, watchNamespace string, recorder record
 	}
 	return err
 }
+
+// markNodeAsUnhealthy marks the given node as unhealthy by adding an annotation to it. If the number of unhealthy nodes
+// exceeds the maximum allowed, an error is returned.
+func markNodeAsUnhealthy(c client.Client, node *core.Node) error {
+	nodeLocker.Lock()
+	defer nodeLocker.Unlock()
+
+	ctx := context.TODO()
+	nodeList := &core.NodeList{}
+	if err := c.List(ctx, nodeList, client.MatchingLabels{core.LabelOSStable: "windows"}); err != nil {
+		return fmt.Errorf("error listing Windows nodes: %w", err)
+	}
+
+	unhealthyCount := 0
+	for _, currentNode := range nodeList.Items {
+		v, ok := currentNode.Annotations[metadata.UnhealthyAnnotation]
+		// check annotation present with value set to true
+		if ok && v == "true" {
+			unhealthyCount++
+			// check if node is already marked as unhealthy
+			if currentNode.Name == node.Name {
+				// node is unhealthy, nothing to do
+				return nil
+			}
+		}
+	}
```

Comment:
You can break this into a separate function which looks for and returns the node with the annotation.

You can then return from this function if the returned node is the same as the node being reconciled.